### PR TITLE
Registering DISTANCE_GRADIENT as a 3D variable with components.

### PR DESCRIPTION
**Pull request by jcotela**
There is a type clash between the DISTANCE_GRADIENT variable in core and a variable with the same name used in an external application. Since this variable used a custom type registration macro, I changed it to use the standard variable with components one, which also solves the type clash problem (the application variable also was a variable-with-components type).

I don't think that this will cause any issues to current users of this variable (namely, the mmg utility), since I haven't changed the underlying type (it is still an `array_1d<double,3>`), just its registration. Just to be safe, @loumalouomega, can you test if the mmg interface is still working fine?

### CODE REVIEW

**Comment by loumalouomega** [comment]
Sure...still waiting for the #385 approval

**Review by loumalouomega — APPROVED** [review]
Everything working properly

**Comment by jcotela** [comment]
@loumalouomega thanks for checking!

**Pull request merged**
